packages/flutter_tools/lib/src/windows/visual_studio_validator.dart

```diff
@@ -30,6 +30,26 @@ class VisualStudioValidator extends DoctorValidator {
           visualStudio.fullVersion,
       )));
 
+      if (visualStudio.isPrerelease) {
+        messages.add(ValidationMessage(userMessages.visualStudioIsPrerelease));
+      }
+
+      // Messages for faulty installations.
+      if (!visualStudio.isComplete) {
```

Comment:
We probably shouldn't add all of these, as I'm not sure how they combine in practice (e.g., it may be that when a reboot required it also reports as incomplete) and we don't want to overwhelm the user with potentially conflicting information. Let's do these three as:
```
if (...isRebootRequired) {
  ...
} else if (!...isComplete) {
  ...
} else if (!...isLaunchable) {
  ...
}
```

And actually, some of the issue reports we've had (see #39970) suggest that if the install isn't fully complete, the components checks don't necessarily work as we would expect which can cause confusion, so let's add the `!...hasNecessaryComponents)` check to the end of that chain as an else-if as well.

Reply:
Done